tcms/rpc/tests/test_testexecutionstatus.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=attribute-defined-outside-init, invalid-name, objects-update-used
+
+from xmlrpc.client import ProtocolError
+from tcms.rpc.tests.utils import APITestCase, APIPermissionsTestCase
+
+
+class TestFilter(APITestCase):
+    """Test TestExecutionStatus.filter"""
+
+    def test_filter_statuses(self):
+        execution_statuses = self.rpc_client.TestExecutionStatus.filter(
+            {"weight__lt": 0}
+        )
+
+        execution_status = execution_statuses[0]
+
+        self.assertGreater(len(execution_statuses), 0)
+        self.assertLess(execution_status['weight'], 0)
+        self.assertGreater(execution_status['id'], 0)
+        self.assertIn(execution_status['name'], ['FAILED', 'ERROR', 'BLOCKED'])
```

Comment:
Above 3 assertions should be executed in a for loop to validate every item in the result, not only the first one.

Here weight has a known value which can be used to validate the filtering works as expected.  The rest of the fields are dynamic but we don't really need their values b/c asserting on weight will be enough to validate the filter operation. 

However you need to assert that each item in the result has all of the other fields present, regardless of their values. Fields are: id, name, color, icon

Reply:
Names can also be translated into other languages and hard-coding them in tests leads to failures as seen in https://github.com/kiwitcms/Kiwi/pull/2043/checks?check_run_id=1327360853

Reply:
roger that, making changes